acceptance-tests/src/test-support/java/tech/pegasys/artemis/test/acceptance/dsl/ArtemisNode.java

```diff
@@ -0,0 +1,187 @@
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.test.acceptance.dsl;
+
+import static org.apache.tuweni.io.file.Files.deleteRecursively;
+import static org.apache.tuweni.toml.Toml.tomlEscape;
+import static org.assertj.core.api.Assertions.assertThat;
+import static tech.pegasys.artemis.util.Waiter.waitFor;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.concurrent.TimeUnit;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.tuweni.bytes.Bytes32;
+import tech.pegasys.artemis.provider.JsonProvider;
+import tech.pegasys.artemis.test.acceptance.dsl.data.BeaconHead;
+
+public class ArtemisNode {
+  private static final Logger LOG = LogManager.getLogger();
+  private static final String ARTEMIS_BINARY =
+      new File(System.getProperty("artemis.binary.path", "../build/install/artemis/bin/artemis"))
+          .getAbsolutePath();
+
+  private final SimpleHttpClient httpClient;
+
+  private boolean started = false;
+  private File configFile;
+  private Process process;
+  private Path workDir;
+
+  ArtemisNode(final SimpleHttpClient httpClient) {
+    this.httpClient = httpClient;
+  }
+
+  public void start() throws Exception {
+    assertThat(started).isFalse();
+    started = true;
+    workDir = Files.createTempDirectory("artemis");
+    configFile = File.createTempFile("config", ".toml");
+    configFile.deleteOnExit();
+    new ConfigBuilder().writeTo(configFile);
+    LOG.info(
+        "Starting artemis from {} with config file {}",
+        ARTEMIS_BINARY,
+        configFile.getAbsolutePath());
+    process =
+        new ProcessBuilder(
+                ARTEMIS_BINARY, "--config", configFile.getAbsolutePath(), "--logging", "DEBUG")
+            .directory(workDir.toFile())
+            .inheritIO()
+            .start();
+  }
+
+  public void waitForGenesis() {
+    waitFor(() -> httpClient.get(getRestApiUrl(), "/node/genesis_time"));
+  }
+
+  public void waitForNewBlock() throws IOException {
+    final Bytes32 startingBlockRoot = getCurrentBeaconHead().getBlockRoot();
+    waitFor(
+        () -> assertThat(getCurrentBeaconHead().getBlockRoot()).isNotEqualTo(startingBlockRoot));
+  }
+
+  private BeaconHead getCurrentBeaconHead() throws IOException {
+    final BeaconHead beaconHead =
+        JsonProvider.jsonToObject(
+            httpClient.get(getRestApiUrl(), "/beacon/head"), BeaconHead.class);
+    LOG.debug("Retrieved beacon head: {}", beaconHead);
+    return beaconHead;
+  }
+
+  private URI getRestApiUrl() {
+    return URI.create("http://127.0.0.1:9051");
```

Comment:
Should you get the port from the ConfigBuilder?  Or have them both pull the port from a constant?

Reply:
Yeah getting from the config builder makes sense.  At some point we'll need to find a way to select unique ports. Although that's looking quite difficult so we may need to use docker...